### ISSUE TITLE
metric for enformer

### ISSUE THREAD

**Issue by Rachel66666** [lucidrains/enformer-pytorch#9]
Hello, can I ask how you find of the human pearson R is 0.625 for validation, and 0.65 for test? Couldn't find any information in the paper. Is there any other place that records this?

**Comment by rnsherpa** [comment]
Sorry for reviving and old thread, but I'd also like to know where these correlation numbers come from with respect to the original paper. It looks like @jstjohn did the correlation analysis [here](https://github.com/lucidrains/enformer-pytorch/blob/main/evaluate_enformer_pytorch_correlation.ipynb). Would be able to shed some light on the question?

**Comment by biginfor** [comment]
I don't think the calculation of Pearson correlation coefficient in the Enformer TensorFlow version is correct (at least in the pipeline they [provided:).](https://colab.research.google.com/github/deepmind/deepmind_research/blob/master/enformer/enformer-training.ipynb) They first calculated the covariance and standard deviation of the true and predicted values within each batch, but did not calculate the Pearson correlation coefficient. Instead, they waited until all batches had calculated the "temporary values" before calculating the Pearson correlation coefficient for the "population". There is no doubt that the "temporary values" are nonlinear, so this will introduce erroneous results.

**Comment by biginfor** [comment]
Let's assume the data are as follows:
**batch1** : input1, target1
**batch2** : input2, target2
**batch3** : input3, target3
The idea behind calculating Pearson correlation coefficient using the original TensorFlow version of the enformer is as follows:
**cor(c(input1,input2,input3),c(target1,target2,target3))**
The idea behind calculating Pearson correlation coefficient using the original Pytorch version of the enformer is as follows:
**mean(cor(input1,target1),cor(input2,target2),cor(input3,target3))**
I think the second option is reasonable.

**Comment by jstjohn** [comment]
Why is the second better? Here each input and target is a different batch along the sequence axis. Taking the global correlation of points is a correlation metric while the mean of subsequence correlations (with arbitrary cut points even) is something else that needs more justification in my opinion. Sent from my iPhoneOn Sep 29, 2024, at 7:40 AM, Eli ***@***.***> wrote:﻿
Let's assume the data are as follows:
batch1 : input1, target1
batch2 : input2, target2
batch3 : input3, target3
The idea behind calculating Pearson correlation coefficient using the original TensorFlow version of the enformer is as follows:
cor(c(input1,input2,input3),c(target1,target2,target3))
The idea behind calculating Pearson correlation coefficient using the original Pytorch version of the enformer is as follows:
mean(cor(input1,target1),cor(input2,target2),cor(input3,target3))
I think the second option is reasonable.

**Comment by biginfor** [comment]
Sorry, I think you're right. Calculating the correlation based on batches and then taking the average is not a good idea as it ignores the global distribution of the data.
![2e3d512f6e1e61a2916bf4584a93309](https://github.com/user-attachments/assets/0d40fab4-22a7-42c2-a8a8-749c7838e298)



> Why is the second better? Here each input and target is a different batch along the sequence axis. Taking the global correlation of points is a correlation metric while the mean of subsequence correlations (with arbitrary cut points even) is something else that needs more justification in my opinion. Sent from my iPhoneOn Sep 29, 2024, at 7:40 AM, Eli ***@***.***> wrote:﻿ Let's assume the data are as follows: batch1 : input1, target1 batch2 : input2, target2 batch3 : input3, target3 The idea behind calculating Pearson correlation coefficient using the original TensorFlow version of the enformer is as follows: cor(c(input1,input2,input3),c(target1,target2,target3)) The idea behind calculating Pearson correlation coefficient using the original Pytorch version of the enformer is as follows: mean(cor(input1,target1),cor(input2,target2),cor(input3,target3)) I think the second option is reasonable. —Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you were mentioned.Message ID: ***@***.***>

**Comment by jstjohn** [comment]
Interesting. I mean the thing that seemed off was that your proposal was over arbitrary cut points. Taking the mean after splitting on a nuisance variable on the other hand could make a ton of sense. That could help control for confounding for example. You could cut the data up by something you don’t want to be included in your correlation measurement. Chromosome boundary for example, maybe GC percent bucket or some other feature you think is a nuisance variable that is not biologically meaningful. Then you could calculate correlation within each group and average. Smaller groups though might be noisier which would make real signals harder to detect. Just some other thoughts!Sent from my iPhoneOn Sep 29, 2024, at 7:07 PM, Eli ***@***.***> wrote:﻿
Sorry, I think you're right. Calculating the correlation based on batches and then taking the average is not a good idea as it ignores the global distribution of the data.
2e3d512f6e1e61a2916bf4584a93309.jpg (view on web)

Why is the second better? Here each input and target is a different batch along the sequence axis. Taking the global correlation of points is a correlation metric while the mean of subsequence correlations (with arbitrary cut points even) is something else that needs more justification in my opinion. Sent from my iPhoneOn Sep 29, 2024, at 7:40 AM, Eli @.> wrote:﻿ Let's assume the data are as follows: batch1 : input1, target1 batch2 : input2, target2 batch3 : input3, target3 The idea behind calculating Pearson correlation coefficient using the original TensorFlow version of the enformer is as follows: cor(c(input1,input2,input3),c(target1,target2,target3)) The idea behind calculating Pearson correlation coefficient using the original Pytorch version of the enformer is as follows: mean(cor(input1,target1),cor(input2,target2),cor(input3,target3)) I think the second option is reasonable. —Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you were mentioned.Message ID: @.>

**Comment by biginfor** [comment]
Thanks! That helps a lot!
